x/nexus/keeper/general_message.go

```diff
@@ -43,14 +43,18 @@ func (k Keeper) SetNewMessage(ctx sdk.Context, m exported.GeneralMessage) error
 		}
 	}
 
-	if _, found := k.getMessage(ctx, m.DestinationChain, m.ID); found {
+	if _, found := k.GetMessage(ctx, m.DestinationChain, m.ID); found {
 		return fmt.Errorf("general message %s for chain %s already exists", m.ID, m.DestinationChain.String())
 	}
+	if err := m.ValidateBasic(); err != nil {
+		return err
+	}
 
-	return k.setMessage(ctx, m)
+	k.getMessagesQueue(ctx, m.DestinationChain).Enqueue(utils.KeyFromStr(getMessageKey(m.DestinationChain, m.ID).String()), &m)
```

Comment:
it assumes the destination is evm, what if the destination is a cosmos chain?
as only evm endblocker dequeue atm